esp/bindings/SOAP/xpp/xpp/XmlPullParser.h

```diff
@@ -608,14 +622,18 @@ namespace xpp {
       }
       int n = stag.attEnd;
       for(int i = 0; i < n; ++i) {
-        if(stag.attArr[i].prefixValid) {
+        if(supportNs && requestNs && ("xmlns" == stag.attArr[i].prefix || "xmlns" == stag.attArr[i].localName)) {
+          stag.attArr[i].uri = stag.attArr[i].value;
+        } else if(stag.attArr[i].prefixValid) {
           SXT_STRING pfx = stag.attArr[i].prefix;
           //if(prefix2Ns[ pfx ]  == NULL)
           if(prefix2Ns.find( pfx ) == prefix2Ns.end())
             throw XmlPullParserException(string(
               "no namespace for attribute prefix ")+pfx
               +tokenizer.getPosDesc(), tokenizer.getLineNumber(), tokenizer.getColumnNumber()); 
           stag.attArr[i].uri = prefix2Ns[ pfx ];
+        } else if(supportNs) {
+          stag.attArr[i].uri = (el.defaultNs ? el.defaultNs : "");
```

Comment:
Is this an unrelated fix?  Or how is it related to exposing the namespace attributes?

Could doing this have unexpected backward compatibility side effects?

Reply:
It was a mistake. It seemed so wrong that an attribute without a prefix would would not belong to the default namespace, when defined. But apparently attributes can only be part of a namespace when prefixed.